model/lists/ListService.php

```diff
@@ -39,6 +39,9 @@
 
 class ListService extends tao_models_classes_ListService
 {
+    /** @var int */
+    private $maxAllowedListElementsLimit = 1000;
```

Comment:
max and limit are the same, so I suggest: 

```
listElementsLimit
```

Methods name will be shorter too. Wdyt?

Reply:
This attribute is matching the name of the parameter from the acceptance criteria in the Jira issue (`max_allowed_list_elements_limit`). The env variable associated with this is `MAX_ALLOWED_LIST_ELEMENTS_LIMIT` in the service provider.

We already have `LOCAL_LIST_ELEMENTS_LIMIT` and `REMOTE_LIST_ELEMENTS_LIMIT` that are used by `ListElementsFinder` to control the no. of items shown without needing to click "Load more". So I think `listElementsLimit` may be confusing (if we'd keep names consistent, the env var for this would be, `LIST_ELEMENTS_LIMIT`, but right now it would only apply to local lists).

To be honest I'm not sure what is the best approach (I already had doubts about this while implementing it). Matching the env var seems "less confusing" but longer. But that would make "limit" to have two meanings: The hard limit of items that can be put on a local list and the number of items fetched from remote ones (afaik).

For this particular instance, maybe we can name the local var `maxListElements` and rename `ListService` methods accordingly (as it is a hard limit for local lists) but keep the env var and constant as `MAX_ALLOWED_LIST_ELEMENTS_LIMIT` to match the AC?

Reply:
The other one is the pagination limit. Could not we rename the previous one to `LOCAL_LIST_ELEMENTS_PAGINATION_LIMIT` ?

The ACs in the ticket are using generic names to help us understanding, but they do not force us to use the same names while writing the code. 

Anyway, I just feel the MAX and LIMIT in the same sentence is a bit redundant, but the ticket can move with this :) 

Thanks for the explanations, @hectoras

Reply:
If renaming things from the ACs and from previous changes can be considered, then, I suggest to:

- Rename old parameters as `LOCAL_LIST_ELEMENTS_PAGINATION_LIMIT` and `REMOTE_LIST_ELEMENTS_PAGINATION_LIMIT`?
  -  hmm... I think `LOCAL_LIST_PAGE_SIZE` / `REMOTE_LIST_PAGE_SIZE` sounds even better :D

- Rename this one to `LOCAL_LIST_MAX_ITEMS` ?

- Rename code vars accordingly (so this one would be `$localListMaxItems`)

I think PAGE_SIZE and MAX_ITEMS are the less confusing (personal preference).

wdyt?

Reply:
If we remain consistent it is not a problem. I would not overload this task if it is finished already. Unless it makes sense for you and we have time, we can do. Otherwise it is ok to keep it like this :)

Reply:
I've renamed them since it was fast to do (I've already spotted places to be changed) :point_right: 8e4f0c3a7b522d202e292d3981ed53a380a1d250.

I think there are no more backend changes for this one :crossed_fingers: